lib/mixlib/shellout.rb

```diff
@@ -297,7 +289,6 @@ def parse_options(opts)
           self.password = setting
         when 'user'
           self.user = setting
-          self.with_logon = setting
```

Comment:
Ohhh, I see, its also an accessor that was public.  Argh...

Yeah, that probably needs to get reverted...

We do *not* want the behavior that setting user automagically sets with_login which causes login shell behavior on unix, however.  That would also be a breaking change.  I think we need to revert back to the original behavior of with_login which is set by default when user is set and only affects the windows side of things.  Then we add the login true/false setting which defaults to falsey and affects the unix side of things.

Then we need to make a breaking change at some point to fix this mess.

I really don't want to bump mixlib-shellout major version again so soon, it causes a lot of chaos.